core/src/test/java/dk/alexandra/fresco/logging/TestArithmeticLoggingDecorators.java

```diff
@@ -0,0 +1,171 @@
+import dk.alexandra.fresco.suite.dummy.arithmetic.DummyArithmeticResourcePool;
+import dk.alexandra.fresco.suite.dummy.arithmetic.DummyArithmeticResourcePoolImpl;
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.hamcrest.core.Is;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class TestArithmeticLoggingDecorators {
+
+  @Test
+  public void testNumericLoggingDecorator() throws Exception {
+    TestThreadRunner.TestThreadFactory<DummyArithmeticResourcePool, ProtocolBuilderNumeric> f
+      = new SqrtTests.TestSquareRoot<>();
+    
+    int noOfParties = 2;
+    EvaluationStrategy evalStrategy = EvaluationStrategy.SEQUENTIAL; 
+    
+    List<Integer> ports = new ArrayList<>(noOfParties);
+    for (int i = 1; i <= noOfParties; i++) {
+      ports.add(9000 + i * (noOfParties - 1));
+    }
+
+    Map<Integer, NetworkConfiguration> netConf =
+        TestConfiguration.getNetworkConfigurations(noOfParties, ports);
+    Map<Integer, TestThreadRunner.TestThreadConfiguration<DummyArithmeticResourcePool, ProtocolBuilderNumeric>> conf =
+        new HashMap<>();
+    BigInteger mod = new BigInteger(
+        "6703903964971298549787012499123814115273848577471136527425966013026501536706464354255445443244279389455058889493431223951165286470575994074291745908195329");
+
+    for (int playerId : netConf.keySet()) {
+      NetworkConfiguration partyNetConf = netConf.get(playerId);
+      
+      DummyArithmeticProtocolSuite ps = new DummyArithmeticProtocolSuite(mod, 200);
+      BatchEvaluationStrategy<DummyArithmeticResourcePool> strat = evalStrategy.getStrategy();
+      ProtocolEvaluator<DummyArithmeticResourcePool, ProtocolBuilderNumeric> evaluator 
+          = new BatchedProtocolEvaluator<>(strat, ps);
+      SecureComputationEngine<DummyArithmeticResourcePool, ProtocolBuilderNumeric> sce 
+          = new SecureComputationEngineImpl<>(ps, evaluator);
+
+      Drbg drbg = new HmacDrbg();
+      TestThreadRunner.TestThreadConfiguration<DummyArithmeticResourcePool, ProtocolBuilderNumeric> ttc =
+          new TestThreadRunner.TestThreadConfiguration<>(sce,
+              () -> new DummyArithmeticResourcePoolImpl(playerId,
+                  noOfParties, drbg, mod),
+              () -> {
+                return new KryoNetNetwork(partyNetConf);
+              });
+      conf.put(playerId, ttc);
+    }
+    TestThreadRunner.run(f, conf);
+    
+    for(Integer pId: netConf.keySet()) {
+      List<PerformanceLogger> pl = DummyArithmeticBuilderFactory.performanceLoggers.get(pId);
+      
+      ListLogger testLogger = new ListLogger();
+      pl.get(0).printToLog(testLogger, pId);
+      String expectedOutput = pl.get(0).makeLogString(pId);
+      Map<String, Object> loggedValues = pl.get(0).getLoggedValues(pId);
+      Assert.assertThat(loggedValues.get(NumericLoggingDecorator.ARITHMETIC_BASIC_ADD), Is.is((long)5719386));
```

Comment:
Did you check that this is these are the correct numbers? I think we ought to test with computations generating more predictable numbers, say a single (few) additions, mults, etc.